WordPress/gutenberg
# Editor: Fix term selector component exports

## What?
Fixes #41661.

PR directly exports `FlatTermSelector` and `HierarchicalTermSelector` components without applying the filter.

## Why?
The developers have to use the same filter to change components used for rendering the taxonomy, causing an infinite loop.

## Testing Instructions
1. Register taxonomy for a post (snippet 1).
2. Open a post.
3. Copy the last example from readme and run it via the DevTools console (snippet 2).
4. Confirm that the "Tracks" taxonomy panel is updated to use the `HierarchicalTermSelector` component.
5. Confirm that there's no infinite loop - the browser doesn't freeze or crash.

```php
add_action( 'init', static function() {
	$args = array(
		'public'            => true,
		'hierarchical'      => false,
		'label'             => 'Tracks',
		'show_ui'           => true,
		'show_in_rest'      => true,
		'show_admin_column' => true,
		'query_var'         => true,
		'rewrite'           => array( 'slug' => 'genre' ),
	);

	register_taxonomy( 'track', array( 'post' ), $args );
} );
```

```js
const el = wp.element.createElement;
const HierarchicalTermSelector = wp.editor.PostTaxonomiesHierarchicalTermSelector;

function customizeTrackSelector( OriginalComponent ) {
	return function ( props ) {
		if ( props.slug === 'track' ) {
			return el( HierarchicalTermSelector, props );
		} else {
			return el( OriginalComponent, props );
		}
	};
}

wp.hooks.addFilter(
	'editor.PostTaxonomyType',
	'my-plugin/set-hierarchical-term-selector',
	customizeTrackSelector
);
```
